pkg/sidecar/k8s_network.go

```diff
@@ -107,12 +107,23 @@ func (n *K8sNetwork) ConfigureNetwork(ctx context.Context, cfg *sync.NetworkConf
 			CapabilityArgs: capabilityArgs,
 		}
 
-		err = retry(3, 2*time.Second, func() error {
-			_, err = n.cninet.AddNetworkList(ctx, netconf, rt)
-			return err
-		})
-		if err != nil {
-			return fmt.Errorf("failed to add network through cni plugin: %w", err)
+		errc := make(chan error)
+
+		go func() {
+			err = retry(3, 2*time.Second, func() error {
+				_, err = n.cninet.AddNetworkList(ctx, netconf, rt)
```

Comment:
I am pretty sure we block here when `weave` goes biserk. I've been seeing the `trying to add a link` log line and nothing after it for a very long time.

Reply:
@raulk it'd be interesting to monitor how long this takes in the `good` case, once we get go-metrics in the codebase.